e2e-tests/utils/index.ts

```diff
@@ -0,0 +1,292 @@
+    debug?: boolean
+  }) {
+    this.taskHandle = spawn(cmd, args, { cwd: opts?.cwd, stdio: 'pipe' });
+
+    if (opts?.debug) {
+      this.taskHandle.stdout.pipe(process.stdout);
+      this.taskHandle.stderr.pipe(process.stderr);
+    }
+  }
+
+  sendStdin(input: string) {
+    this.taskHandle.stdin.write(input);
+  }
+
+  waitForOutput(output: string, timeout = 240_000, optional = false) {
+    return new Promise<boolean>((resolve, reject) => {
+      let outputBuffer = '';
+      const timeoutId = setTimeout(() => {
+        if (optional) {
+          // The output is not found but it's optional so we can resolve the promise with false
+          resolve(false);
+        } else {
+          reject(new Error(`Timeout waiting for output: ${output}`));
+        }
+      }, timeout);
+
+      this.taskHandle.stdout.on('data', (data) => {
+        outputBuffer += data;
+        if (outputBuffer.includes(output)) {
+          clearTimeout(timeoutId);
+          // The output is found so we can resolve the promise with true
+          resolve(true);
+        }
+      });
+    });
+  }
+
+  kill() {
+    this.taskHandle.kill('SIGINT');
+  }
+}
+
+/**
+ * Initialize a git repository in the given directory
+ * @param projectDir
+ */
+export function initGit(projectDir: string): void {
+  try {
+    execSync('git init', { cwd: projectDir });
+    // Add all files to the git repo
+    execSync('git add -A', { cwd: projectDir });
+    // Add author info to avoid git commit error
+    execSync('git config user.email test@test.sentry.io', { cwd: projectDir });
+    execSync('git config user.name Test', { cwd: projectDir });
+    execSync('git commit -m init', { cwd: projectDir });
+  } catch (e) {
+    log.error('Error initializing git');
+    throw e;
+  }
+}
+
+/**
+ * Cleanup the git repository in the given directory
+ *
+ * Caution! Make sure `projectDir` is a test project directory,
+ * if in doubt, please commit your local non-test changes first!
+ * @param projectDir
+ */
+export function cleanupGit(projectDir: string): void {
+  try {
+    // Remove the .git directory
+    execSync(`rm -rf ${projectDir}/.git`);
+  } catch (e) {
+    log.error('Error cleaning up git');
+    throw e;
+  }
+}
+
+/**
+ * Revert local changes in the given directory
+ *
+ * Caution! Make sure `projectDir` is a test project directory,
+ * if in doubt, please commit your local non-test changes first!
+ *
+ * @param projectDir
+ */
+export function revertLocalChanges(projectDir: string): void {
+  try {
+    // Revert tracked files
+    execSync('git checkout .', { cwd: projectDir });
+    // Revert untracked files
+    execSync('git clean -fd .', { cwd: projectDir });
+  } catch (e) {
+    log.error('Error reverting local changes');
+    throw e;
+  }
+}
+
+/**
+ * Run the wizard with the given integration and project directory
+ *
+ * @param integration
+ * @param projectDir
+ */
+export async function runWizard(integration: Integration, projectDir: string) {
```

Comment:
What if, instead of having all of this functionality in this function, we call this function something like `startWizardInstance` and it returns the `wizardTestEnv` so you can call stuff on it inside the jest tests? That way, we're a little more dynamic in what we can test and the various permutations we can test for more complicated wizards.

Reply:
Updated and moved to Jest runner 👍